packages/flutter/lib/src/gestures/recognizer.dart

```diff
@@ -324,9 +352,17 @@ abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecogni
   @override
   void handleEvent(PointerEvent event) {
     assert(state != GestureRecognizerState.ready);
-    if (state == GestureRecognizerState.possible && event.pointer == primaryPointer) {
-      // TODO(abarth): Maybe factor the slop handling out into a separate class?
-      if (event is PointerMoveEvent && _getDistance(event) > kTouchSlop) {
+    if (event.pointer == primaryPointer) {
+      final bool isPreAcceptSlopPastTolerance =
+          state == GestureRecognizerState.possible
+          && preAcceptSlopTolerance != null
+          && _getDistance(event) > preAcceptSlopTolerance;
+      final bool isPostAcceptSlopPastTolerance =
+          state == GestureRecognizerState.accepted
+          && postAcceptSlopTolerance != null
+          && _getDistance(event) > postAcceptSlopTolerance;
+
+      if (event is PointerMoveEvent && (isPreAcceptSlopPastTolerance || isPostAcceptSlopPastTolerance)) {
         resolve(GestureDisposition.rejected);
```

Comment:
I don't understand. If `isPostAcceptSlopPastTolerance` is true, then we know we've already been accepted, so what does it mean to reject it?

Reply:
I don't have a strong opinion about this. I'm mostly leaning towards not introducing more API breaks in this PR since it was the existing behaviour (whether the gesture was accepted or not). Rejecting an accepted gesture makes future events not go through, such as the up event.

Reply:
I don't understand how your comment corresponds to my question.

Reply:
> If `isPostAcceptSlopPastTolerance` is true, then we know we've already been accepted

Yes

> so what does it mean to reject it?

In the existing implementation, 
```dart
if (event is PointerMoveEvent && _getDistance(event) > kTouchSlop) {
  resolve(GestureDisposition.rejected);
```
causes the recognizer to stop emitting post-accept events such as the up event. I leaned towards not altering that behavior and left it in this PR though we can make that change if we feel that we have a strong reason to.